### DISCUSSION TITLE
Can’t start SWS

### DISCUSSION THREAD

**Discussion by Geobert** [static-web-server/static-web-server#326] [Q&A] [answered]
with this toml:
```
[general]
host = "::"
port = 443
root = "/home/geob/www/blog/public"
page404 = "/home/geob/www/blog/public/404.html"
http2 = true
http2-tls-cert = "/etc/letsencrypt/live/new_domain.com/cert.pem"
http2-tls-key = "/etc/letsencrypt/live/new_domain.com/privkey.pem"
https-redirect = true

[advanced]

[[advanced.redirects]]
host = "https://old_domain.com"
source = "/{*}"
destination = "https://new_domain.com/$1"
kind = 301
```

when starting I get this error:
```
Mar 07 18:18:20 geob-server static-web-server[101323]: 2024-03-07T17:18:20.889589Z  INFO static_web_server::info: static-web-server 2.27.0
Mar 07 18:18:20 geob-server static-web-server[101323]: 2024-03-07T17:18:20.889750Z  INFO static_web_server::info: log level: error
Mar 07 18:18:20 geob-server static-web-server[101323]: 2024-03-07T17:18:20.889998Z ERROR static_web_server::server: server failed to start up: failed to bind to [::]:80 address
Mar 07 18:18:20 geob-server static-web-server[101323]: Caused by:
Mar 07 18:18:20 geob-server static-web-server[101323]:     Address family not supported by protocol (os error 97)
Mar 07 18:18:20 geob-server systemd[1]: sws.service: Main process exited, code=exited, status=1/FAILURE
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ An ExecStart= process belonging to unit sws.service has exited.
░░
░░ The process' exit code is 'exited' and its exit status is 1.
Mar 07 18:18:20 geob-server systemd[1]: sws.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ The unit sws.service has entered the 'failed' state with result 'exit-code'.
```

I don’t understand why it tries to bind to port 80 as the toml specified port 443

**Comment by Geobert** [comment]
and in my systemd unit file:
`ExecStart=/usr/local/bin/static-web-server -w /home/geob/sws.toml`

**Comment by Geobert** [comment]
I use a toml file in order to have advanced section for redirecting from old domain

**Comment by joseluisq** [comment, marked as answer]
If you want to sandbox SWS using systemd then you have to use a file descriptor instead of binding to a tcp port. Follow https://static-web-server.net/features/file-descriptor-socket-passing/ for more details.

And to change the server port used by the service, edit the value of `ListenStream` in the `static-web-server.socket` file example.

**Comment by Geobert** [comment]
Thanks for your answer!
I’m now facing a strange issue: ` path /home/geob/www/blog/public was not found or inaccessible`
But the path does exist, and everythings under `www` is in group `www-data` (as I have `SupplementaryGroups=www-data`) 
I tried with `User=root` or `User=geob` with no luck

What am I missing this time?

EDIT: can’t access home, moved to /var/ and it works now, thank you again!

**Comment by aaronpeacock** [comment]
WHY is this NOT in the documentation???
"You are NOT free to use any folder you like, even if it's owned by the user and group running SWS, you MUST place your folder in /var"
This info should be added to the documentation!
